fhir-persistence/src/main/java/com/ibm/fhir/persistence/FHIRPersistence.java

```diff
@@ -72,8 +90,26 @@
      *         an OperationOutcome with hints, warnings, or errors related to the interaction
      * @throws FHIRPersistenceException
      */
+    @Deprecated
     <T extends Resource> SingleResourceResult<T> update(FHIRPersistenceContext context, String logicalId, T resource) throws FHIRPersistenceException;
 
+    /**
+     * Updates an existing FHIR Resource by storing a new version in the datastore. This implementation
+     * is added to replace the deprecated {@link #update(FHIRPersistenceContext, String, Resource)} method
+     * This new method expects the resource being passed in to already be modified with correct
+     * meta information. It no longer updates the meta itself. The newVersionId parameter is expected
+     * to match the value in the resource meta value.
+     *
+     * @param context the FHIRPersistenceContext instance associated with the current request
+     * @param logicalId the logical id of the FHIR Resource to be updated
+     * @param newVersionId the new version number to use for the resource
```

Comment:
we could just get this from the resource meta and then we wouldn't need to deprecate the old flavor of this method